### PR TITLE
Story/11523 - Fix failing EDI Unit Tests

### PR DESCRIPTION
**Pull request by PeterC10**

### CODE REVIEW

**Review by squah-u — APPROVED** [review]

File: addons/edi_sale_forward_request/tests/test_sale_forward_request.py
Comment (nocturnalastro) [inline addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: action_execute shouldn't throw errors

**Review by davidmcgaw — APPROVED** [review]
The changes look ok to me but please get the tests running in Travis before merging

File: addons/edi_sale_forward_request/tests/test_sale_forward_request.py
Comment (PeterC10) [reply addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: No but `action_execute` will call `action_prepare` if the EDI doc isn't already prepared. I just kept it as `action_execute` as that what was in the test before but I can change this if preferred. :)

Comment (nocturnalastro) [reply addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: action_prepare shouldn't throw an error either. Thats why we have issues.

Comment (nocturnalastro) [reply addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: Theres a ```except Exception as err```  
https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi/models/edi_document.py#L347

**Comment by nocturnalastro** [comment]
I don't see and TravisCI running is it not setup for this repo?

**Comment by PeterC10** [comment]
@nocturnalastro Yes it's not currently running on TravisCI which is why the unit test failures were not picked up until recently when @davidmcgaw was auditing the tests on the repos. I'm in the process of getting this added now and won't merge this until the checks are running.

File: addons/edi_sale_forward_request/tests/test_sale_forward_request.py
Comment (PeterC10) [reply addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: Isn't that why we have `assertRaisesIssue` though? So it can mute issues that come up when loading the EDI.

For example here:
https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi/tests/test_edi_document.py#L37

File: addons/edi/models/edi_partner_record.py
Comment (nocturnalastro) [inline addons/edi/models/edi_partner_record.py]: Why is this needed?

Comment (PeterC10) [reply addons/edi/models/edi_partner_record.py]: There are tests in `edi_sale` which reference this `parent_id` field that previously didn't exist in this repo.

This field was added in by another custom module at some point, but I don't know why it wasn't added in one of the EDI modules originally.

`parent_id` seemed generic enough to me for it to be included in `edi` and not just `edi_sale`.

Comment (PeterC10) [reply addons/edi/models/edi_partner_record.py]: Sorry I should have been be more clear, the tests fail because `remove_sales_for_invalid_partner_updates` in `edi_sale` references this `parent_id` field.

https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi_sale/models/edi_sale_request_document.py#L127

Comment (nocturnalastro) [reply addons/edi/models/edi_partner_record.py]: Wouldn't the correct thing to do to the only override that method in the custom module otherwise we just end up bloating edi_partner_record. I know its only one field but a parent of a partner is not really part of most partners.

File: addons/edi_sale_forward_request/tests/test_sale_forward_request.py
Comment (nocturnalastro) [reply addons/edi_sale_forward_request/tests/test_sale_forward_request.py:60]: Ah sorry I miss read that as ```assertRaises(```

File: addons/edi/models/edi_partner_record.py
Comment (nocturnalastro) [reply addons/edi/models/edi_partner_record.py]: Refactor this out into an `get_invalid_partners` then its easier
https://github.com/unipartdigital/odoo-edi/blob/master/addons/edi_sale/models/edi_sale_request_document.py#L126

Comment (PeterC10) [reply addons/edi/models/edi_partner_record.py]: Yeah good idea, will go look into that. :)

**Comment by PeterC10** [comment]
CI up and running. :)

**Review by adamrpatrick — APPROVED** [review]

**Pull request merged**
